skyportal/handlers/api/observation_plan.py

```diff
@@ -982,3 +992,349 @@ async def get(self, localization_id, telescope_id):
+          - in: query
+            name: maximumPhase
+            nullable: true
+            schema:
+              type: number
+            description: |
+              Maximum phase (in days) post event time to consider detections. Defaults to 3.
+          - in: query
+            name: injectionFilename
+            nullable: true
+            schema:
+              type: string
+            description: |
+              Path to file for injestion as a simsurvey.models.AngularTimeSeriesSource. Defaults to nsns_nph1.0e+06_mejdyn0.020_mejwind0.130_phi30.txt from https://github.com/mbulla/kilonova_models.
+        responses:
+          200:
+            content:
+              application/json:
+                schema: SingleObservationPlanRequest
+        """
+
+        number_of_injections = self.get_query_argument("numberInjections", 1000)
+        number_of_detections = self.get_query_argument("numberInjections", 1)
+        detection_threshold = self.get_query_argument("detectionThreshold", 5)
+        minimum_phase = self.get_query_argument("minimumPhase", 0)
+        maximum_phase = self.get_query_argument("maximumPhase", 3)
+        injection_filename = self.get_query_argument(
+            "injectionFilename",
+            'data/nsns_nph1.0e+06_mejdyn0.020_mejwind0.130_phi30.txt',
+        )
+
+        options = [
+            joinedload(ObservationPlanRequest.observation_plans)
+            .joinedload(EventObservationPlan.planned_observations)
+            .joinedload(PlannedObservation.field)
+        ]
+
+        observation_plan_request = ObservationPlanRequest.get_if_accessible_by(
+            observation_plan_request_id,
+            self.current_user,
+            mode="read",
+            raise_if_none=True,
+            options=options,
+        )
+        self.verify_and_commit()
+
+        allocation = Allocation.get_if_accessible_by(
+            observation_plan_request.allocation_id,
+            self.current_user,
+            raise_if_none=True,
+        )
+
+        localization = (
+            Localization.query_records_accessible_by(
+                self.current_user,
+            )
+            .filter(Localization.id == observation_plan_request.localization_id)
+            .first()
+        )
+
+        instrument = allocation.instrument
+
+        if instrument.sensitivity_data is None:
+            return self.error('Need sensitivity_data to evaluate efficiency')
+
+        observation_plan = observation_plan_request.observation_plans[0]
+        planned_observations = observation_plan.planned_observations
+        num_observations = observation_plan.num_observations
+        if num_observations == 0:
+            return self.error('Need at least one observation to evaluate efficiency')
+
+        unique_filters = observation_plan.unique_filters
+
+        if not set(unique_filters).issubset(set(list(instrument.sensitivity_data))):
```

Comment:
Should this be `set(d['filter'] for d in instrument.sensitivity_data)`?

Reply:
OK I was wrong. But maybe more explicit would be `set(instrument.sensitivity_data.keys())`?

Reply:
yes